ci/statistics/databases.py

```diff
@@ -0,0 +1,671 @@
+
+        if match:
+            index = int(match.group(1))
+        else:
+            raise SystemError("Cannot convert foldername")
+
+        return index
+
+
+class MeasurementMetadata(MeasurementBase):
+    """Class to represent a table of measurement metadata"""
+
+    def __init__(self, lake, name, data_amount, data_length, client, testmode):
+
+        super().__init__(lake, name, data_amount, data_length, client, testmode)
+        self.array = np.zeros((self.size, 7), dtype=np.int32)
+        self.row_id = 0
+
+        self.array = []
+        self.client = client
+
+        # TODO move to baseclass
+        self.database = f"sturdy-mechanic-312713.ADataSet.{self.name}"
+
+    def scrap_data(self, file: str) -> Tuple[str, str, str, str, str]:
+        """Read measurement data from file"""
+
+        with open(file) as content:
+            data = json.load(content)
+            run = data["run_number"]
+            date = data["updated_at"]
+            url = data["html_url"]
+            name = data["name"]
+            branch = data["head_branch"]
+
+        return run, date, url, name, branch
+
+    def postprocess(self, folders):
+        """Postprocess all relevant folders"""
+        idx = 0
+        for folder in folders:
+            index = self.foldername_to_index(folder)
+
+            name = f"system_test_{index}_metadata.json"
+            path = os.path.join(self.lake, name)
+
+            run, date, url, name, branch = self.scrap_data(path)
+
+            self.array.append((idx, run, date, url, name, branch))
+            idx += 1
+
+        return self.array
+
+    def show(self):
+        """Show content on console"""
+        logging.info("Content of table %s", self.database)
+        for row in self.array:
+            logging.info(row)
+
+    def update_table(self):
+        """Create table and prepare loading via json and upload"""
+
+        logging.info("Updating table: %s", self.name)
+
+        self.delete_table()
+
+        self.job_config = bigquery.LoadJobConfig(
+            schema=[
+                bigquery.SchemaField("id", "INT64"),
+                bigquery.SchemaField("mid", "INT64"),
+                bigquery.SchemaField("date", "STRING"),
+                bigquery.SchemaField("url", "STRING"),
+                bigquery.SchemaField("name", "STRING"),
+                bigquery.SchemaField("branch", "STRING"),
+            ],
+        )
+
+        self.json_data = []
+
+        for index in range(self.data_amount):
+            self.json_data.append(
+                {
+                    "id": self.array[index][0],
+                    "mid": self.array[index][1],
+                    "date": self.array[index][2],
+                    "url": self.array[index][3],
+                    "name": self.array[index][4],
+                    "branch": self.array[index][5],
+                }
+            )
+
+        self.upload_table()
+
+
+class CpuHistory(MeasurementBase):
```

Comment:
Same thing here: I miss some content description: 1) the purpose 2) the schema 3) is this a source dataset or a derived one.